data/transactions/application.go

```diff
@@ -119,22 +126,29 @@ type ApplicationCallTxnFields struct {
 	// the app or asset id).
 	Accounts []basics.Address `codec:"apat,allocbound=encodedMaxAccounts"`
 
+	// ForeignAssets are asset IDs for assets whose AssetParams
+	// (and since v4, Holdings) may be read by the executing
+	// ApprovalProgram or ClearStateProgram.
+	ForeignAssets []basics.AssetIndex `codec:"apas,allocbound=encodedMaxForeignAssets"`
+
 	// ForeignApps are application IDs for applications besides
 	// this one whose GlobalState (or Local, since v4) may be read
 	// by the executing ApprovalProgram or ClearStateProgram.
 	ForeignApps []basics.AppIndex `codec:"apfa,allocbound=encodedMaxForeignApps"`
 
+	// Access unifies `Accounts`, `ForeignApps`, `ForeignAssets`, and `Boxes`
+	// under a single list. It removes all implicitly available resources, so
+	// "cross-product" resources (holdings and locals) must be explicitly
+	// listed, as well as app accounts, even the app account of the called app!
+	// Transactions using Access may not use the other lists.
+	Access []ResourceRef `codec:"al,allocbound=encodedMaxAccess"`
```

Comment:
A comment on each of the other lists stating / referring to the above may be useful (though there are plenty of references to this fact throughout the code).

Ideally, no code ourside of this file would have to know about tx.Access vs the various slices, but I con't think that is true atm.